pegen/pegen.c

```diff
@@ -2,6 +2,22 @@
 #include "pegen.h"
 #include "v38tokenizer.h"
 
+int _set_context(expr_ty e, expr_context_ty ctx)
+{
+    // TODO: Check for valid names and the rest
+    // of the possibilities in the e->kind enum.
+    // Also, research a way to propagate errors.
+    switch (e->kind) {
+        case Name_kind:
+            e->v.Name.ctx = ctx;
```

Comment:
I'm not sure, if this applies here, but @gvanrossum suggests [here](https://github.com/gvanrossum/pegen/issues/79#issuecomment-559611541) that modifying something in place could mess up the cache.

Reply:
Yes I think it applies here. 

PS I am on a phone at a dentist office, code reviews are tough.

Reply:
Thanks for pointing that out. I have used the new helper `store_name` in f8f48c4